src/types/misc.ts

```diff
@@ -348,3 +348,18 @@ export interface SortOptions {
 export interface MenuMouseEvent extends MouseEvent {
   closedMenuId?: UID;
 }
+
+// https://github.com/Microsoft/TypeScript/issues/13923#issuecomment-557509399
+// prettier-ignore
+export type Immutable<T> =
```

Comment:
deserves a dedicated commit imho, it's really nice but people should be aware of its existence to somehow use it